pkg/azure/auth.go

```diff
@@ -40,6 +41,12 @@ func getAuthorizer(authLocation string, useManagedidentity bool, cpConfig *Cloud
 	if !useManagedidentity && cpConfig != nil {
 		klog.V(1).Info("Creating authorizer using Cluster Service Principal.")
 		credAuthorizer := auth.NewClientCredentialsConfig(cpConfig.ClientID, cpConfig.ClientSecret, cpConfig.TenantID)
+
+		// Set active directory endpoint using environment
+		azureEnv, _ := azure.EnvironmentFromName(cpConfig.Cloud)
+		credAuthorizer.AADEndpoint = azureEnv.ActiveDirectoryEndpoint
```

Comment:
does NewClientCredentialsConfig have any constructor which allows u to initialize endpoint directly?

Reply:
no other constructor that takes clientId and secret as an input. There is one that takes from environment.